cip-ledger-state.md

```diff
@@ -0,0 +1,111 @@
+complexity for chain following applications like db-sync.
+
+# Current Situation
+
+Currently there is a local chain sync protocol which is really just the peer-to-peer protocol
+using a local domain socket rather than the TCP/IP socket normally used for P2P transport.
+
+The data transported over this local chain sync protocol is limited to block chain data. However, a
+Cardano node also maintains ledger state which includes:
+
+* The current UTxO state.
+* Current amount of ADA delegated to each stake pool.
+* Which stake address is currently delegated to each pool.
+* Rewards account balances for each stake address.
+* Current protocol parameters.
+
+The first of these ledger state components is by far the largest component and is probably not
+needed outside the node (and definitely not needed by db-sync). However the others are needed and
+stored by `cardano-db-sync` which gets these data sets by maintaining its own copy of ledger state
+and periodically extracting the parts required.
+
+This means that when `node` and `db-sync` are run on the same machine (which is the recommended
+configuration) that machine has two basically identical copies of ledger state. Ledger state is a
+*HUGE* data structure and the mainnet version currently consumes over 10 Gigabytes of memory.
+Furthermore, maintaining ledger state duplicates code functionality that is in `ouroboros-consensus`
+and the maintenance of ledger state has been the cause of about 90% of the bugs in `db-sync` over
+the last two years. The maintenance of ledger state also makes updating, running and maintaining
+`db-sync` by operators more difficult than it should be. Finally, if `db-sync` did not have to
+maintain ledger state, the size of the `db-sync` code base would probably decrease by about 50% and
+the bits removed are some of the most complicated parts.
+
+
+## Specification
+
+The proposed solution is an enhanced local chain sync protocol that would only be served over a
+local domain socket. The enhanced chain chain sync protocol would include the existing block chain
+data as well as events containing the ledger data that db-sync needs. This enhanced local chain
+sync protocol is useful for many applications other than just db-sync.
+
+Smart contract developers would like an application that turns block chain and ledger state changes
+into an event stream. With this enhanced local chain sync protocol, generating an easily consumable
+event stream simply requires a conversion of the binary enhanced local chain sync protocol into
+JSON.
+
+This enhanced local chain sync protocol is basically the data that would be provided by the
+proposed Scientia program (which as far as I am aware has been dropped).
+
+The ledger state data that would provided over the extended chain sync protocol is limited to:
+
+* Per epoch stake distribution (tickle fed in a deterministic manner).
+* Per epoch rewards (tickle fed in a deterministic manner).
```

Comment:
```suggestion
* Per epoch stake distribution (trickle fed in a deterministic manner).
* Per epoch rewards (trickle fed in a deterministic manner).
```
I can't find any other references to "tickle fed" so I'm guessing this is what we are talking about? https://dictionary.cambridge.org/dictionary/english/trickle-feed